src/vm/method.hpp

```diff
@@ -2702,18 +2721,29 @@ class NDirectMethodDesc : public MethodDesc
         ndirect.m_dwECallID = dwID;
     }
 
+    // In DAC builds this function is meant to be called only for structures, which are not part of other structures
+    PTR_CUTF8 GetLibNameRaw()
+    {
+        LIMITED_METHOD_DAC_CONTRACT;
+
+        return ndirect.m_pszLibName.GetValueMaybeNull(dac_cast<TADDR>(this) + offsetof(NDirectMethodDesc, ndirect.m_pszLibName));
```

Comment:
The existing uses of relative pointers seem to be using `RelativePointer::GetValueAtPtr/GetValueMaybeNullAtPtr` to avoid referencing the field twice. I think it is a good pattern to follow because of it avoids mismatches between the two references. Using the existing pattern, this line would be:

`RelativePointer<PTR_CUTF8>::GetValueMaybeNullAtPtr(PTR_HOST_MEMBER_TADDR(NDirectMethodDesc, this, ndirect.m_pszLibName))`

What do you think?